c7n/resources/rds.py

```diff
@@ -281,6 +300,56 @@ def process(self, dbs):
                 **params)
 
 
+@filters.register('upgrade-available')
+class UpgradeAvailable(Filter):
+    """ Scan DB instances for available engine upgrades
+
+        This will pull DB instances & check their specific engine for any
+        engine version with higher release numbers than the current one
+
+        This will also tag the rds instance with 'target_engine' which is
+        the most recent version of the engine available
+    """
+
+    schema = type_schema('upgrade-available', value={'type': 'boolean'})
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('rds')
+        results = []
+        engine_upgrades = _list_engines_upgrade_version(client)
+        for r in resources:
+            upgrades = engine_upgrades[r['Engine']]
+            if len(upgrades) == 0 or r['EngineVersion'] not in upgrades:
+                continue
+            target_upgrade = upgrades[r['EngineVersion']]
+            r['c7n.rds-minor-engine-upgrade'] = target_upgrade
+            if self.data.get('value', True):
+                if target_upgrade != r['EngineVersion']:
+                    if not r in results:
```

Comment:
this if/else block can be simplified
```
op = self.data.get('value', True) and operator.ne or operator.eq
if op(r, r['EngineVersion):
   results.append(r)
```
also its unclear why its trying to pseudo unique the resources (aka if not r in results)